lib/Sema/CSSimplify.cpp

```diff
@@ -3646,6 +3659,19 @@ ConstraintSystem::matchExistentialTypes(Type type1, Type type2,
     return getTypeMatchFailure(locator);
   }
 
+  // ConformsTo constraints are generated when opening a generic
+  // signature with a RequirementKind::Conformance requirement, so
+  // we must handle pack types on the left by splitting up into
+  // smaller constraints.
+  if (auto *packType = type1->getAs<PackType>()) {
```

Comment:
Why isn't this handled by `simplifyConformsToConstraint` that accepts a `Type`?

Reply:
There are two overloads, one with a Type and one with a ProtocolDecl... maybe we could collapse them into one though?

Reply:
I think it makes sense to add it to the first one only, that's what is going to be called for constraint simplification.